src/authenticate.js

```diff
@@ -0,0 +1,86 @@
+import Sketch from 'sketch';
+import FormData from 'sketch-polyfill-fetch/lib/form-data';
+
+import {SECRET_NAME, TOKEN_NAME, API_ENDPOINT, CLIENT_ID} from './config';
+
+/**
+ * Ask the user for the secret and attempt to authenticate
+ */
+export default () => {
+  Sketch.UI.getInputFromUser(
+    'Please enter the token beginning "mnz"',
+    {
+      initialValue: '',
+      description: 'This is intended for internal use only',
+    },
+    async (err, value) => {
+      if (err) return;
+
+      const res = await authenticate(value);
+
+      const {access_token} = res;
+
+      if (access_token) {
+        Sketch.Settings.setSettingForKey(TOKEN_NAME, access_token);
+        Sketch.Settings.setSettingForKey(SECRET_NAME, value);
+        Sketch.UI.message('🗝 Token saved');
+      } else {
+        Sketch.UI.message('🤔 Hmm we had a problem authenticating you');
+      }
+    }
+  );
+};
+
+/**
+ * Token exchange
+ */
+export const authenticate = async secret => {
+  const body = new FormData();
+  body.append('grant_type', 'client_credentials');
+  body.append('client_id', CLIENT_ID);
+  body.append('client_secret', secret);
+
+  const res = await fetch(`${API_ENDPOINT}/oauth2/token`, {
+    method: 'POST',
+    body: body,
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+      Accept: 'application/json',
+    },
+  });
+
+  if (!res.ok) {
+    await refresh();
+  }
+
+  return await res.json();
+};
+
+/**
+ * Refresh the token
+ */
+export const refresh = async () => {
+  const secret = Sketch.Settings.settingForKey(SECRET_NAME);
+  const refreshToken = Sketch.Settings.settingForKey(TOKEN_NAME);
+
+  const body = new FormData();
+  body.append('grant_type', 'refesh_token');
+  body.append('client_id', CLIENT_ID);
+  body.append('refresh_token', refreshToken);
+  body.append('client_secret', secret);
+
+  const res = await fetch(`${API_ENDPOINT}/oauth2/token`, {
```

Comment:
could you pull most of this out into a function that you can call from both `authenticate` and `refresh` as thet seem v similar?